lib/openshift-rest-client.js

```diff
@@ -18,100 +18,50 @@
  *
  */
 
-const privates = require('./private-map');
-const loadConfig = require('./config-loader');
+const zlib = require('zlib');
+const fs = require('fs');
+const path = require('path');
 
-const buildconfigs = require('./build-configs');
-const builds = require('./builds');
-const clusterrolebindings = require('./cluster-role-bindings');
-const configmaps = require('./configmaps');
-const deploymentconfigs = require('./deployment-config');
-const deployments = require('./deployments');
-const events = require('./events');
-const groups = require('./groups');
-const imagestreams = require('./imagestreams');
-const imagestreamtags = require('./imagestreamtags');
-const ingress = require('./ingress');
-const persistentvolumeclaims = require('./persistent-volume-claims');
-const pods = require('./pods');
-const projectrequests = require('./projectrequests');
-const projects = require('./projects');
-const replicationcontrollers = require('./replication-controllers');
-const rolebindings = require('./role-bindings');
-const routes = require('./routes');
-const secrets = require('./secrets');
-const services = require('./services');
-const serviceinstances = require('./serviceinstances');
-const statefulsets = require('./statefulsets');
+const { Client, config, alias } = require('kubernetes-client');
 
-function bindModule (client, input) {
-  if (typeof input === 'object') {
-    const initializedModule = {};
-    for (const name in input) {
-      initializedModule[name] = bindModule(client, input[name]);
-    }
-    return initializedModule;
-  } else if (typeof input === 'function') {
-    return input(client);
-  } else {
-    throw new Error(`Unexpected input module type: ${input}`);
+// A list of shorthand aliases
+const resourceAliases = {
+  'apps.openshift.io': ['app'],
+  'authorization.openshift.io': ['authorization'],
+  'build.openshift.io': ['build'],
+  'image.openshift.io': ['image'],
+  'network.openshift.io': ['network'],
+  'oauth.openshift.io': ['oauth'],
+  'project.openshift.io': ['project'],
+  'quota.openshift.io': ['quota'],
+  'route.openshift.io': ['route'],
+  'security.openshift.io': ['security'],
+  'template.openshift.io': ['template'],
+  'user.openshift.io': ['user']
+};
+
+// function is passed in and called by the kubernetes-client to add the openshift aliases
+function getNames (resourceType) {
+  const aliases = [resourceType];
+  if (resourceAliases[resourceType]) {
+    return aliases.concat(resourceAliases[resourceType]);
   }
+  return alias(resourceType);
 }
 
+// unzip and load the openshift openapi swagger spec file
+// Doesn't look like there is a way to query this from a running cluster
+// Perhaps, we could try to query https://raw.githubusercontent.com/openshift/origin/master/api/swagger-spec/openshift-openapi-spec.json ?
```

Comment:
If you do decide to take this route, I think you'd want to not query `master` but instead target a specific version.

Reply:
right, if we did do this,  it would be in another PR/version i think